src/d3a/models/market/two_sided.py

```diff
@@ -264,9 +277,70 @@ def match_recommendation(self, recommended_list):
                     recommended_list, index+1, trade, bid_trade
                 )
 
+    @staticmethod
+    def validate_requirements_satisfied(obj1: Union[Offer, Bid], obj2: Union[Offer, Bid]) -> None:
+        """Validate if both trade parties satisfy each other's requirements.
+
+        :raises:
+            InvalidBidOfferPair: Bid offer pair failed the validation
+        """
+        def _validate_requirements_satisfied(_obj1: Union[Offer, Bid], _obj2: Union[Offer, Bid]):
+            if _obj1.requirements:
+                for requirement in _obj1.requirements:
+                    is_satisfied = True
+                    for key, value in requirement.items():
+                        if (key == "preferred_trading_partner" and
+                                not ((hasattr(_obj2, "buyer_origin_id") and
+                                      _obj2.buyer_origin_id in value) or
+                                     (hasattr(_obj2, "buyer_id") and
+                                      _obj2.buyer_id in value) or
+                                     (hasattr(_obj2, "seller_origin_id") and
+                                      _obj2.seller_origin_id in value) or
+                                     (hasattr(_obj2, "seller_id") and
+                                      _obj2.seller_id in value))):
+                            assert isinstance(value, list)
+                            is_satisfied = False
+                            continue
+                        if (key == "energy_type" and
+                                (not _obj2.attributes or
+                                 _obj2.attributes.get("energy_type") not in value)):
+                            assert isinstance(value, list)
+                            is_satisfied = False
+                            continue
+                        if (key == "hashed_identity" and
+                                (not _obj2.attributes or
+                                 _obj2.attributes.get("hashed_identity") != value)):
+                            assert isinstance(value, str)
+                            is_satisfied = False
+                            continue
+
+                        if is_satisfied:
+                            """Requirement dict is satisfied."""
+                            return
```

Comment:
It is important that True is returned here, right?
```suggestion
                            return True
```

Reply:
Ignore me, understood now.